File: week11/week11-1.py
# 開新的 +程式碼 Python
# week11-1.py 主題 Hash Map, Hash Set (在「資料結構與演算法」教過的 Hash 雜湊)
table = {} # Python用大括號,代表字典、也就是 Hash Table (Hash Map, Hash Set)
table[9977341] = 9
table[4433997] = 4

print(table[9977341]) # 會印出 9
print(table[4433997]) # 會印出 4
# print(table[0]) # 糟! 如果沒有這個key, 就會爆炸!
if 0 not in table: # 如果沒有在裡面
  print("table[0]不存在")

File: week11/week11-1.cpp
/// week11-1.cpp 使用 Hash Map 或 Hash Set
/// 不能直接在 CodeBlocks 使用, 因為是 C++ 2021
/// Setting-Compiler, 加入 c++11 就可以了
#include <iostream>
#include <unordered_map>
using namespace std;

int main()
{
    unordered_map<int,int> table;
    table[9977341] = 9;
    table[4433997] = 4;
    cout << table[9977341] << "\n";
    cout << table[4433997] << "\n";
}
